### PR TITLE
CD: add simple CD based on GitHubActions

### PR DESCRIPTION
**Pull request by YozhikVolkoff**
Adds workflow 'packaging.yml', based on GitHub Actions. 
Workflow starts on push or pull request.

Fixes bug when CMake tried to found C++ compiler (not needed) by
adding 'C' in project() statement in CMakeLists.txt files. 
CMakeLists.txt file in Msgpuck submodule was updated 
in the same way for this reason too.
Note: this submodule update contains irrelevant changes.

Due to some troubles on Ubuntu's site, Groovy was excluded 
from packaging matrix.

### CODE REVIEW

File: .github/workflows/packaging.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          - { os: 'fedora', dist: '34' }
+          - { os: 'opensuse-leap', dist: '15.1' }
+          - { os: 'opensuse-leap', dist: '15.2' }
+          - { os: 'ubuntu', dist: 'xenial' }
+          - { os: 'ubuntu', dist: 'bionic' }
+          - { os: 'ubuntu', dist: 'focal' }
+          - { os: 'ubuntu', dist: 'hirsute' }
+          # Skip packaging for Ubuntu Groovy due to 
+          # some troubles on Ubuntu's site
+          # - { os: 'ubuntu', dist: 'groovy' }
```
Comment (Totktonada) [inline .github/workflows/packaging.yml:36]: I would not add it at all, neither uncommented, nor commented: this distro is in the EOL state: https://wiki.ubuntu.com/Releases.

**Comment by Totktonada** [comment]
Please, highlight in the commit message the reason of the msgpuck submodule updating (`project(msgpuck C)` change) and note that this submodule update contains irrelevant changes (which looks safe, but anyway).

File: CMakeLists.txt [hunk 2]
```diff
@@ -21,8 +21,8 @@ if (CMAKE_C_COMPILER_ID STREQUAL Clang)
     set(CMAKE_COMPILER_IS_GNUCXX OFF)
 endif()
 
-# set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread -Wall -Wextra -std=gnu99 -D__USE_XOPEN")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -std=gnu99 -D__USE_XOPEN")
+# set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread -Wall -Wextra -std=gnu99 -D__USE_XOPEN -fPIC")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -std=gnu99 -D__USE_XOPEN -fPIC")
```
Comment (Totktonada) [inline CMakeLists.txt:25]: > Fixes the compilation without -fPIC flag, which caused error during
> preparing packages for Ubuntu Xenial. This flag was added
> in set(CMAKE_C_FLAGS ...) statement in CMakeLists.txt.

Can you reveal more info about the problem? I tried to build a package without this change for Ubuntu Xenial and it succeeds. What also confuses me: AFAIR, cmake enables `-fPIC` for shared libraries by default.

And I would also link [this](https://stackoverflow.com/q/38296756/1598057) StackOverflow question about most idiomatic way to enforcing position independent code (if we really need to set it explicitly).

File: .github/workflows/packaging.yml (hunk 1 above)
Comment (YozhikVolkoff) [reply .github/workflows/packaging.yml]: Fixed.

File: CMakeLists.txt (hunk 2 above)
Comment (YozhikVolkoff) [reply CMakeLists.txt]: AFAIR, I tried packaging for Xenial without -fPIC flag and it didn't succeed. I've removed this flag and have tried again – now it passes.

File: .github/workflows/packaging.yml (hunk 1 above)
Comment (Totktonada) [reply .github/workflows/packaging.yml]: I meant exactly 'don't add'. Now it is added and then removed.

File: CMakeLists.txt (hunk 2 above)
Comment (Totktonada) [reply CMakeLists.txt]: Okay so.

File: .github/workflows/packaging.yml (hunk 1 above)
Comment (YozhikVolkoff) [reply .github/workflows/packaging.yml]: Okay. I deleted the second commit and rewrote the first.

File: .github/workflows/packaging.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+name: packaging
+
+on: [push, pull_request]
+
+jobs:
+  biuld_package:
```
Comment (Totktonada) [inline .github/workflows/packaging.yml:6]: Just observed a typo: biuld -> build.

**Review by Totktonada — APPROVED** [review]
LGTM.

File: .github/workflows/packaging.yml (hunk 3 above)
Comment (YozhikVolkoff) [reply .github/workflows/packaging.yml]: Thanks. Fixed.

**Review by Korablev77 — APPROVED** [review]

**Pull request merged**